diff --git a/libc/libc/inet/socketcalls.c uClibc-0.9.32-noaccept4/libc/inet/socketcalls.c
--- a/libc/inet/socketcalls.c	2011-06-08 12:35:20.000000000 -0700
+++ b/libc/inet/socketcalls.c	2011-06-08 20:44:04.000000000 -0700
@@ -9,6 +9,7 @@
 #include <errno.h>
 #include <syscall.h>
 #include <sys/socket.h>
+#include <bits/kernel-features.h>
 
 #ifdef __NR_socketcall
 extern int __socketcall(int call, unsigned long *args) attribute_hidden;
@@ -85,6 +86,7 @@ libc_hidden_weak(accept)
 
 #ifdef L_accept4
 #ifdef __NR_accept4
+#ifdef __ASSUME_ACCEPT4
 # define __NR___sys_accept4  __NR_accept4
 static _syscall4(int, __sys_accept4, int, fd, struct sockaddr *, addr, socklen_t *, addrlen, int, flags)
 int accept4(int fd, struct sockaddr *addr, socklen_t * addrlen, int flags)
@@ -122,6 +124,7 @@ int accept4(int fd, struct sockaddr *add
 }
 #endif
 #endif
+#endif
 
 #ifdef L_bind
 #ifdef __NR_bind
